proxy/proxy.go

```diff
@@ -54,7 +56,11 @@ const (
 // director is a custom internal function for altering a http.Request
 func director(o *options) func(*http.Request) {
 	return func(r *http.Request) {
-		c, err := o.hostMapper(r.Context(), r)
+		ctx := r.Context()
+		newCtx, span := otel.GetTracerProvider().Tracer("").Start(ctx, "x.proxy")
```

Comment:
It is better to overwrite the variable, as you don't want to, by accident, use the original context ever again.
```suggestion
		ctx, span := otel.GetTracerProvider().Tracer("").Start(ctx, "x.proxy")
```